# Lab Instructions

Find a dataset that interests you. I'd recommend starting on [Kaggle](https://www.kaggle.com/). Read through all of the material about the dataset and download a .CSV file.

1. Write a short summary of the data.  Where did it come from?  How was it collected?  What are the features in the data?  Why is this dataset interesting to you?  

2. Identify 5 interesting questions about your data that you can answer using Pandas methods.  

3. Answer those questions!  You may use any method you want (including LLMs) to help you write your code; however, you should use Pandas to find the answers.  LLMs will not always write code in this way without specific instruction.  

4. Write the answer to your question in a text box underneath the code you used to calculate the answer.



In [5]:
import pandas as pd

df = pd.read_csv(r'C:\Users\spoll\Desktop\Full Sail\Data Viz\Data_Visualization_And_Modeling_Online\Lab\social media influencers - instagram.csv', encoding='ISO-8859-1')

# Clean up column names
df.columns = df.columns.str.strip()
print(df.head())

  Influencer insta name             instagram name          category_1  \
0                   433                        433  Sports with a ball   
1          __youngbae__                    TAEYANG               Music   
2           _agentgirl_  НАСТЯ ИВЛЕЕВА               Shows   
3           _imyour_joy                        Joy           Lifestyle   
4         _jeongjaehyun                    Jaehyun                 NaN   

  category_2 Followers Audience country(mostly) Authentic engagement  \
0        NaN     48.5M                    Spain               383.1K   
1        NaN     12.7M                Indonesia                 478K   
2        NaN     18.8M                   Russia               310.8K   
3        NaN     13.5M                Indonesia                 1.1M   
4        NaN     11.1M                Indonesia                 2.5M   

  Engagement avg  
0           637K  
1         542.3K  
2         377.9K  
3           1.4M  
4           3.1M  


This dataset contains information on the top 1,000 social media influencers on Instagram. It was collected by Kaggle user Ramjas Maurya and compiled from publicly available Instagram data. The dataset includes features such as the influencer's username, their display name, their primary and secondary content categories, their follower count, their audience's most common country, their authentic engagement, and their average engagement. I chose this dataset because I'm interested in content creation and social media, and I want to understand what separates top influencers from each other especially by category and country.

Which content category has the most influencers?

In [7]:
print(df['category_1'].value_counts().head(10))

category_1
Music                        235
Cinema & Actors/actresses    214
Sports with a ball           113
Lifestyle                     99
Shows                         43
Modeling                      36
Beauty                        28
Humor & Fun & Happiness       18
Art/Artists                   12
Family                        11
Name: count, dtype: int64


The most common content category among the top 1,000 Instagram influencers is Music with 235 influencers, followed closely by Cinema & Actors/Actresses with 214. This makes sense as celebrities in entertainment naturally build massive followings. Lifestyle (99) and Sports (113) also rank high, showing that aspirational content dominates the platform.

In [ ]:
Which country has the most influencers?

In [15]:
print(df['Audience country(mostly)'].value_counts().head(10))

Audience country(mostly)
United States     279
Brazil            152
India             149
Indonesia         134
Mexico             45
Spain              34
Russia             24
Argentina          23
Iran               23
United Kingdom     20
Name: count, dtype: int64


The country with the most top Instagram influencers is the United States, with 279 influencers, followed by Brazil (152) and India (149). The US dominance reflects the outsized role American pop culture plays in global social media. It's also notable that Indonesia ranks 4th with 134 influencers, showing that Southeast Asia has a major and growing force in the influencer space.

Who are the top 10 influencers by follower count?

In [9]:
df['Followers_clean'] = df['Followers'].str.replace('M', 'e6').str.replace('K', 'e3').astype(float)
top10 = df.nlargest(10, 'Followers_clean')[['instagram name', 'category_1', 'Followers']]
print(top10)

         instagram name                 category_1 Followers
388           Instagram                Photography    487.2M
203   Cristiano Ronaldo         Sports with a ball    419.6M
510          Kylie 🤍                    Fashion    323.3M
532           Leo Messi         Sports with a ball    315.4M
814        Selena Gomez                      Music    308.2M
899             therock  Cinema & Actors/actresses      307M
87        Ariana Grande                      Music    302.3M
494      Kim Kardashian                    Fashion    296.4M
126           Beyoncé                      Music    246.9M
489  Khloé Kardashian         Clothing & Outfits    230.2M


The most followed influencer in this dataset is Instagram's official account with 487.2M followers, followed by Cristiano Ronaldo (419.6M) and Kylie Jenner (323.3M). The top 10 is dominated by athletes, musicians, and actors,  reflecting that traditional celebrity status translates directly into social media reach.

In [ ]:
Which category has the highest average engagement?

In [11]:
df['Engagement_clean'] = df['Engagement avg'].str.strip().str.replace('M', 'e6').str.replace('K', 'e3').astype(float)
print(df.groupby('category_1')['Engagement_clean'].mean().sort_values(ascending=False).head(10))

category_1
Management & Marketing       2.050000e+06
Fashion                      2.003344e+06
Music                        1.066516e+06
Business & Careers           1.022575e+06
Lifestyle                    8.181535e+05
Clothing & Outfits           7.906200e+05
Cinema & Actors/actresses    7.895888e+05
Modeling                     7.590750e+05
Photography                  6.958000e+05
Sports with a ball           6.610637e+05
Name: Engagement_clean, dtype: float64


Despite Music having the most influencers, Management & Marketing has the highest average engagement at approximately 2.05M, followed by Fashion at 2.0M. This suggests that smaller, more niche categories drive stronger audience interaction even with fewer total influencers, quality over quantity.

In [ ]:
How many influerncer audiences are primarily based in the US?

In [13]:
us_influencers = df[df['Audience country(mostly)'] == 'United States']
print(f"Number of influencers with US audiences: {len(us_influencers)}")
print(us_influencers[['instagram name', 'category_1', 'Followers']].head(10))

Number of influencers with US audiences: 279
      instagram name                 category_1 Followers
10               NaN                      Music     13.2M
13           50 Cent                      Music     27.9M
22      Adam Sandler  Cinema & Actors/actresses     12.7M
23               NaN                     Beauty     40.5M
24             Adele                      Music     49.7M
25  adidas Originals         Clothing & Outfits     34.2M
26               NaN                        NaN      2.6M
34               NaN                        NaN     10.8M
37        Alexis Ren                  Lifestyle     15.8M
53    Amaury Guichon             Food & Cooking      6.7M


279 out of 1,000 top Instagram influencers have audiences primarily in the United States, making it by far the most represented country. This reflects the US being the largest English-speaking market on the platform and highlights how American pop culture drives global social media trends.